lib/repositories/records/mongo_db.rb

```diff
@@ -0,0 +1,105 @@
+# frozen_string_literal: true
+
+require 'connections/mongo_db'
+require 'sport_type'
+
+module Repositories
+  module Records
+    class MongoDb
+      attr_accessor :years, :sport_types, :group_by
+
+      def distance_per_month(years:, sport_types:)
+        q = [{ '$match' => matcher(years:, sport_types:) }]
+        q << { '$group' => {
+          _id:      { year: '$year', month: '$month' },
+          distance: { '$sum' => '$distance' }
+        } }
+        q << { '$sort' => { distance: -1 } }
+        q << { '$limit' => LIMIT }
+
+        data = sessions.aggregate(q)
+        data.each_with_object({}) do |doc, h|
+          key = "#{doc['_id']['year']}-#{doc['_id']['month']}"
+          h[key] = doc['distance'] / 1000.0
+        end
+      end
+
+      def distance_per_week(years:, sport_types:)
+        q = [{ '$match' => matcher(years:, sport_types:) }]
+        q << { '$group' => {
+          _id:      { year: '$year', month: '$month', week: { '$isoWeek' => '$start_time' } },
+          distance: { '$sum' => '$distance' }
+        } }
+        q << { '$sort' => { distance: -1 } }
+        q << { '$limit' => LIMIT }
+
+        data = sessions.aggregate(q)
+        data.each_with_object({}) do |doc, h|
+          key = "#{doc['_id']['year']}-#{doc['_id']['month']}-#{doc['_id']['week']}"
```

Comment:
Why the `month` is in the key? Is the year+week_number not enough?

Reply:
Hmmm ... yes .. also the query above has the month in it .. which would lead to an issue maybe or at least looks strange if a week crosses a month .. or would create two entries for the same week basically.. 

removed month at the key + the query. 

Good catch !